client/websocket.js

```diff
@@ -99,6 +99,46 @@ class WebsocketClient extends EventEmitter {
     beginConnection.call(this);
   }
 
+  ping(pingText = "") {
+    const message = Buffer.from(pingText);
+    const len = message.length;
+    const frameHeader = Buffer.from([0x89]);
```

Comment:
This and the pong method have heavy repeat code - copied from the send method. Should really abstract to a common method later.